src/x/cache/lru_cache.go

```diff
@@ -430,10 +430,15 @@ func (c *LRU) handleCacheLoadErrorWithLock(
 // updateCacheEntryWithLock updates a cache entry with a new value or cached error,
 // and marks it as the most recently accessed and most recently loaded entry
 func (c *LRU) updateCacheEntryWithLock(
-	key string, expiresAt time.Time, value interface{}, err error,
+	key string, expiresAt time.Time, value interface{}, err error, enforceLimit bool,
 ) (interface{}, error) {
 	entry := c.entries[key]
 	if entry == nil {
+		if enforceLimit && c.reserveCapacity(1) != nil {
```

Comment:
nit: maybe check the returned error type is `ErrCacheFull` in case a future contributor changes `reserveCapacity` to return some other kind of error. or change the signature of `reserveCapacity` to return a bool instead to be more explicit it's not really an error.

Reply:
Indeed, replacing return value `error` with `bool` for `reserveCapacity` makes lots of sense. Should have thought of this myself.